Added config page

renjithis opened the pull request:
Hi,
I have added a config page (currently hosted in my dropbox Public folder) which will enable the user to change the date format and battery hide interval

![pebble-screenshot_2015-01-18_21-12-49](https://cloud.githubusercontent.com/assets/1526467/5792962/e3ab2d8a-9f56-11e4-87df-2e91a75015b2.png)

renjithis committed:
Added config sample

Need to set config URL and create page

renjithis committed:
Added a config file

renjithis committed:
Pointing to config file

renjithis committed:
Fixed duplicate names in dateFormatter enum

renjithis committed:
Fixed typedef in dateFormatter enum

renjithis committed:
Added battery hide interval to config page. TODO: app side

renjithis committed:
Added JSON parsing

renjithis committed:
changed - to _ for JSON keys

renjithis committed:
Wrong format for function pointer in app_config.h. commented for now

renjithis committed:
settings change

renjithis committed:
Added autoconfig submodule. will switch to another branch for integrating it

renjithis committed:
Set basic autoconf. Need to read settings in app and modify behaviour accodringly

renjithis committed:
Config for date format and batter hide time done. Known issue: have to wait a minute for proper display update.
TODO: Invert colours
.
TODO : Optimizations

renjithis committed:
Added copyright disclaimer and handled hide_battery_interval=100 -> infinity

renjithis committed:
Cosmetic changes : Removed a few newlines and spaces

tgaurnier commented:
Unfortunately I don't have much time right now to merge and test and submit to the Pebble store and all that, I encourage you to fork it into your own watchface and customize it to your liking as much as you want.

Also, I advise against having the config page on your Drop Box account, they have a limit how many times a file can be accessed in a day, if too many people use the watchface and access the config they'll start seeing error messages, you can host it right off of GitHub. The address to access the HTML from Github is as follows:
https://rawgithub.com/username/ProjectName/master/path_to_html_file.html

renjithis committed:
Moved to minute monitoring instead of seconds. Should improve battery life (according to Pebble team)

renjithis commented:
Thanks for the advice. I will move it GitHub.
I am adding a few optimizations as well (most has been added to the fork, need to test it a bit more thoroughly). I should be able to submit the completed code by the end of this week. Hopefully you can spare some time to review it and merge.
Since Pebble has added auto pebble app update in the current beta android app (and it should be enabled in the next release), updating the existing app made more sense.

Pebble autoconfig has an embeded mode in which they pass the entrire HTML in the openURL function, but that, by their own admission is a hack and may not work on all phones

renjithis committed:
Unsubscribe from seconds tick after hiding battery indicator. Added a bit of DEBUG logging. possible bug which prevents battery from hiding after interval

tgaurnier commented:
I'll try to get some time to work on it, honestly I think it's kinda cool you found my little app worth contributing to, and I definitely need experience working with others.

Keep in mind I have been planning on adding configuration to this app for quite a while, but I have a global configuration I was going to use, so if I merge your changes I will eventually replace the HTML with my global configuration HTML. To see what I'm talking about check out my Weather Face app, ignore the configuration file in it's repo (I just forgot to delete it a long time ago), it's using my Global Pebble Config which you'll find here: https://github.com/tgaurnier/PebbleGlobalConfig

It works by passing a JSon object defining the HTML form for the Pebble app. 

I do this so all my apps I make for Pebble will have the same feel/theme for the configuration page, also so I can bang out a configuration faster.

renjithis committed:
Replaced dropbox html hostong to github

renjithis committed:
Switched to using timer for hiding the battery. cant subscribe to 2 types of tick time services at the same time

renjithis committed:
Increases size of date format array and string

renjithis committed:
Move date layer a bit to the right when battery is shown so as not to hide the day

renjithis committed:
Update appinfo.json

Changed url from githubusercontent.com to rawgit.com since the other one was showing html source

renjithis committed:
Added animation to move date when displaying and hiding battery

renjithis committed:
Added Inverter layer to reflet the Inver Colours setting

renjithis committed:
Cosmetic changes : removed some commeted code

renjithis committed:
Disabled text move animation on battery show/hide for now

renjithis committed:
Added hourly vibrate setting to the app

renjithis committed:
Drew BT / icon when bt disconnects

renjithis committed:
Forgot to copy autoconf.html from build to config

renjithis committed:
Added hourly vibrate start/stop hours setting

renjithis committed:
Updated autoconfig.html. BUg found in autoconfig - multiple number slider in a page: slider move doesn't change number value
